Src/Sarif.PatternMatcher.Sdk/ValidatorBase.cs

```diff
@@ -53,9 +55,32 @@ protected ValidatorBase()
             PerFileFingerprintCache = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
         }
 
+        protected virtual string ScanIdentityGuid
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(scanIdentityGuid))
```

Comment:
>IsNullOrEmpty [](http://example.com/codeflow?start=27&length=13)

according to this code, you can never set scanIdentityGuid to an empty value. i.e., a null check here would be just fine. So I'd be tempted to render this code as a two-liner:

scanIdentityGuid ??= Guid.NewGuid().ToString();
return scanIdentityGuid;